src/TLuaInterpreter.cpp

```diff
@@ -5484,222 +5483,256 @@ int TLuaInterpreter::tempTimer(lua_State* L)
+        triggerID = pLuaInterpreter->startTempLineTrigger(from, howMany, QString());
+
+        auto trigger = host.getTriggerUnit()->getTrigger(triggerID);
+        trigger->mRegisteredAnonymousLuaFunction = true;
+        lua_pushlightuserdata(L, trigger);
+        lua_pushvalue(L, 3);
+        lua_settable(L, LUA_REGISTRYINDEX);
+    } else {
+        lua_pushfstring(L, "tempLineTrigger: bad argument #3 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 1));
         lua_error(L);
         return 1;
-    } else {
-        luaFunction = lua_tostring(L, 3);
     }
 
-    Host& host = getHostFromLua(L);
-    TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
-    QString _fun = luaFunction.c_str();
-    int timerID = pLuaInterpreter->startTempLineTrigger(luaFrom, luaTo, _fun);
-    lua_pushnumber(L, timerID);
+    lua_pushnumber(L, triggerID);
     return 1;
 }
 
-// tempTrigger( string name, string regex, string function to call, multiline, fg, bg, filter, match all(perlSlashG), highlight,
-// play sound, fire length(mStayOpen), lineDelta).
 int TLuaInterpreter::tempComplexRegexTrigger(lua_State* L)
 {
+    Host& host = getHostFromLua(L);
+    TLuaInterpreter* pLuaInterpreter = host.getLuaInterpreter();
     bool multiLine, matchAll, highlight, playSound, filter, colorTrigger;
     int fireLength, lineDelta;
     QString fgColor, bgColor;
     QStringList regexList;
-    QString script, parent, pattern, soundFile;
+    QString parent, script, triggerName, pattern, soundFile;
     QColor hlFgColor, hlBgColor;
+    QList<int> propertyList;
+    int triggerID;
+
     if (!lua_isstring(L, 1)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #1 type (trigger name create or add to as string expected, got %s!)", luaL_typename(L, 1));
         lua_error(L);
         return 1;
-    } else {
-        parent = lua_tostring(L, 1);
-    }
-
-    if (!lua_isstring(L, 2)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isstring(L, 2)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #2 type (regex pattern to match as string, got %s!)", luaL_typename(L, 2));
         lua_error(L);
         return 1;
-    } else {
-        pattern = lua_tostring(L, 2);
-    }
-    if (!lua_isstring(L, 3)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isstring(L, 3) && !lua_isfunction(L, 3)) {
+        lua_pushfstring(L, "tempComplexRegexTrigger: bad argument #3 type (code to run as a string or a function expected, got %s!)", luaL_typename(L, 3));
         lua_error(L);
         return 1;
-    } else {
-        script = lua_tostring(L, 3);
-    }
-    if (!lua_isnumber(L, 4)) {
-        lua_pushstring(L, "tempComplexRegexTrigger: wrong argument type");
+    } else if (!lua_isnumber(L, 4)) {
```

Comment:
Aren't some of these `boolean` arguments - is there any reason to handle them as numbers - especially in the error messages? 😕

Reply:
They are. That's how @Chris7 did them originally though.